service/history/workflow/mutable_state_impl.go

```diff
@@ -2090,6 +2093,20 @@ func (ms *MutableStateImpl) DeleteSignalRequested(
 	ms.approximateSize -= len(requestID)
 }
 
+func (ms *MutableStateImpl) attachRequestID(requestID string, eventType enumspb.EventType) {
```

Comment:
Set `executionStateUpdated` to true in this func so that the closeTransaction logic can update lastUpdateVersionedTransition for execution state.